### PR TITLE
Add terminal velocities to P3 scheme

### PR DESCRIPTION
**Pull request by trontrytel**
Continuing from https://github.com/CliMA/CloudMicrophysics.jl/pull/334

### CODE REVIEW

File: src/P3Scheme.jl [hunk 1]
```diff
@@ -451,10 +427,191 @@ function distribution_parameter_solver(
...
+    bₙₛ_m(b) = bₙₛ(b) + p3.β_va
+
+    non_spheres_n(a, b, c) = (aₙₛ(a), bₙₛ(b), cₛ(c))
+    non_spheres_m(a, b, c) = (aₙₛ_m(a), bₙₛ_m(b), cₛ(c))
+
+    aᵣₛ(a) = aₛ(a) * (p3.ρ_i / ρ_g)^(2 * κ)
+    aᵣₛ_m(a) = aᵣₛ(a) * FT(π) / 6 * ρ_g
+
+    rimed_n(a, b, c) = (aᵣₛ(a), bₙₛ(b), cₛ(c))
+    rimed_m(a, b, c) = (aᵣₛ_m(a), bₙₛ_m(b), cₛ(c))
```
Comment (anastasia-popova) [inline src/P3Scheme.jl:502]: same here but b_s_m

File: Project.toml [hunk 2]
```diff
@@ -7,6 +7,7 @@ version = "0.18.0"
 ClimaParams = "5c42b081-d73a-476f-9059-fd94b934656c"
 DocStringExtensions = "ffbed154-4ef7-542d-bbb7-c09d3a79fcae"
 ForwardDiff = "f6369f11-7733-5829-9624-2563aa707210"
+QuadGK = "1fd47b50-473d-5c70-9696-f719f8f3bcdc"
```
Comment (nefrathenrici) [inline Project.toml:10]: Where is QuadGK used? I couldn't find any references

Comment (trontrytel) [reply Project.toml:10]: In the last integral for the P3 terminal velocities: https://github.com/CliMA/CloudMicrophysics.jl/blob/0945be50c95cecdcefc8561635e4db65305454c4/src/P3Scheme.jl#L621

File: docs/src/P3Scheme.md [hunk 3]
```diff
@@ -159,3 +159,37 @@ Using this approach we get the following relative errors for ``\lambda``
 include("plots/P3LambdaErrorPlots.jl")
 ```
 ![](P3LambdaHeatmap.png)
+
+## Terminal Velocity
+
+We calculated moment-weighted terminal velocity speeds using the [Chen2022](@cite) velocity parametrization.
```
Comment (anastasia-popova) [inline docs/src/P3Scheme.md:165]: Chen is not getting mentioned here, only cited

**Review by anastasia-popova — APPROVED** [review]
Looks good to me! There is a small mistake I made in the docs where I cited Chen 2022 but did not mention it but should be a quick fix.

File: Project.toml [hunk 4]
```diff
@@ -22,13 +23,14 @@ CloudyExt = "Cloudy"
...
 [compat]
-ClimaParams = "0.10"
+ClimaParams = "0.10.5"
 Cloudy = "0.4"
 DataFrames = "1.6"
 DocStringExtensions = "0.8, 0.9"
 EnsembleKalmanProcesses = "1.1.5"
 ForwardDiff = "0.10"
 MLJ = "0.20"
+QuadGK = "2.9.4"
```
Comment (nefrathenrici) [inline Project.toml:33]: Does the compat need to be so precise?

File: src/P3Scheme.jl [hunk 5]
```diff
@@ -288,6 +231,54 @@ function DSD_N₀(p3::PSP3, N::FT, λ::FT) where {FT}
...
+
+ - x₀ - lower bound
+ - x_end - upper bound
+ - c1, c2, c3 - respective constants
+
+f(D, c1, c2, c3) = c1 * D ^ (c2) * exp(-c3 * D)
+
+Integrates f(D, c1, c2, c3) dD from x₀ to x_end
+"""
+function ∫_Γ(x₀::FT, x_end::FT, c1::FT, c2::FT, c3::FT) where {FT}
```
Comment (nefrathenrici) [inline src/P3Scheme.jl:245]: This is a great function name!

File: docs/Project.toml [hunk 6]
```diff
@@ -9,6 +9,7 @@ DocumenterCitations = "daee34ce-89f3-4625-b898-19384cb65244"
 Literate = "98b081ad-f1c9-55d3-8b20-4c87d4299306"
 OrdinaryDiffEq = "1dea7af3-3e70-54e6-95c3-0bf5283fa5ed"
 Plots = "91a5bcdd-55d7-5caf-9e0b-520d859cae80"
+QuadGK = "1fd47b50-473d-5c70-9696-f719f8f3bcdc"
```
Comment (nefrathenrici) [inline docs/Project.toml:12]: This dependency should only be needed in the main Project.toml, unless you use it in the docs directly. The same applies for the test Project.toml

**Review by nefrathenrici — APPROVED** [review]
Very readable code, thank you!

File: Project.toml (hunk 4 above)
Comment (trontrytel) [reply Project.toml:33]: Probably not. - Lets see!

Comment (trontrytel) [reply Project.toml:33]: But in a separate PR

**Pull request merged**